### PR TITLE
Print strings like `"abc"` instead of `AsList 3 "abc"`, matching the way we parse them.

### PR DESCRIPTION
**Pull request by dougalm**

### CODE REVIEW

**Review by axch — APPROVED** [review]
LGTM, but why did the Github actions not run the Haskell test suite (only Julia)?

**Comment by dougalm** [comment]
> LGTM, but why did the Github actions not run the Haskell test suite (only Julia)?

I think it only runs the Haskell tests when you're proposing to merge into main, but this is proposing to merge into "fenceposts" since it's a follow-up on that.

File: src/lib/Serialize.hs [hunk 1]
```diff
@@ -90,24 +90,26 @@ getTableElements tab = do
 -- This doesn't handle parentheses well. TODO: treat it more like PrettyPrec
 prettyVal :: (MonadIO1 m, EnvReader m, Fallible1 m) => Val n -> m n (Doc ann)
 prettyVal val = case val of
+  -- Pretty-print strings
+  DataCon "AsList" _ [Word8Ty] _ [_, tab@(TabVal _ _)] -> do
+    atoms <- getTableElements tab
```
Comment (apaszke) [inline src/lib/Serialize.hs:95]: Can you please use `getDexString` here? It's _a lot_ faster

**Review by apaszke — APPROVED** [review]
Nice!

File: src/lib/Serialize.hs (hunk 1 above)
Comment (dougalm) [reply src/lib/Serialize.hs:95]: Good idea

**Pull request merged**